### PR TITLE
Added loading and handling of object definitions

### PR DESCRIPTION
**Pull request by tlf30**
Hello, I have added loading in objects such as trees, rocks, etc. 
To do this, I created a ObjectDefinition and ObjectDefinitionParser, I kept them as similar as possible to the others for Items and NPCs. Then I added them to the launchPool in the Server. 

I created a obj_defs.json using a program I wrote to extract the data from the cache that is packaged with the Asteria 2.0 client you provided me. If there is other data we need from the cache regarding objects, let me know and I can modify me program to extract it. 

If this is not how you want to do this, please let me know. But this seemed like the next step forward in getting this data into the server. 

Thanks,
~Trevor

### CODE REVIEW

**Comment by tlf30** [comment]
Sorry for the large number of commits in this one!

**Comment by tlf30** [comment]
I found some data that is corrupt when pulled from cache. I fixed it by hand but I need to find out if the cache is corrupt or if my program is not reading it correctly. The strange part is that it was only a handful of objects that had the issue out of the few thousand of them, and only in the descriptions.

File: src/main/java/io/luna/game/model/def/ObjectDefinition.java [hunk 1]
```diff
@@ -0,0 +1,215 @@
+package io.luna.game.model.def;
+
+import com.google.common.collect.ImmutableList;
+import io.luna.util.StringUtils;
+import io.luna.util.parser.impl.ObjectDefinitionParser;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+/**
+ *
```
Comment (lare96) [inline src/main/java/io/luna/game/model/def/ObjectDefinition.java]: "A definition model describing ..."

File: src/main/java/io/luna/game/model/def/ObjectDefinition.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+import io.luna.util.StringUtils;
+import io.luna.util.parser.impl.ObjectDefinitionParser;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+/**
+ *
+ * @author Trevor Flynn {@literal <trevorflynn@liquidcrystalstudios.com>}
+ */
+public class ObjectDefinition {
```
Comment (lare96) [inline src/main/java/io/luna/game/model/def/ObjectDefinition.java]: This should be final

File: src/main/java/io/luna/game/model/def/ObjectDefinition.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+    /**
+     * Returns an iterable containing all definitions.
+     */
+    public static Iterable<ObjectDefinition> all() {
+        return DEFINITIONS;
+    }
+
+    /**
+     * Returns the non-player name of {@code id}.
```
Comment (lare96) [inline src/main/java/io/luna/game/model/def/ObjectDefinition.java]: object** name

**Comment by lare96** [comment]
Other then the above it looks great and will merge once updated. Great work man

**Comment by tlf30** [comment]
I fixed the documentation and made the class final.

**Pull request merged**